gpt_bpe_test.go

```diff
@@ -1069,7 +1071,7 @@ func TestModelDownloadLlama(t *testing.T) {
 	_, rsrcErr := resources.ResolveResources(modelId, destPathPTR,
 		resources.RESOURCE_MODEL, rsrcType, hfApiToken)
 	if rsrcErr != nil {
-		//os.RemoveAll(destPath)
+		os.RemoveAll(destPath)
```

Comment:
Instead of putting all these `os.RemoveAll(destPath)` everywhere for cleanup, why not just add a `defer os.RemoveAll(destPath)`?

As long as `destPath` does change after it's instantiated, I'd just suggest putting it here:

```
func TestModelDownloadLlama(t *testing.T) {
	// Pythia uses a slightly different file structure, where
	// the vocab.json and merges.txt files are stored in the
	// tokenizer.json file. We want to check if we are able to
	// download the model and extract the vocab.json and merges.txt
	modelId := "georgesung/llama2_7b_chat_uncensored"
	destPath := "./TestModelDownloadLlama"
	destPathPTR := &destPath
        defer os.RemoveAll(destPath)
```

Along with any other cleanup you do. You don't need the initial `os.RemoveAll` if cleanup is doing its job.